concordium-consensus/src/Concordium/MultiVersion.hs

```diff
@@ -1228,7 +1228,7 @@ startBaker MultiVersionRunner{mvBaker = Nothing, ..} =
         Runner
         LLError
         "Attempted to start baker thread, but consensus was started without baker credentials."
-startBaker mvr@MultiVersionRunner{mvBaker = Just Baker{..}, ..} = do
+startBaker mvr@MultiVersionRunner{mvBaker = Just Baker{..}, ..} = whenBakerRequired $ do
```

Comment:
Consider adding to the doc comment above that this function is only interesting for older protocol versions and does nothing for newer versions